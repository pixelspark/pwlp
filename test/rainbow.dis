0000.	e0	USER	get_length
0001.	20	PEEKB	0
0002.	11	PUSHB	[01]
0004.	81	BINARY	SUB
0005.	10	PUSHB	0
0006.	e3	USER	set_pixel
0007.	01	POP	1
0008.	71	UNARY	DEC
0009.	60	JNZ	to 1
0012.	01	POP	1
0013.	e4	USER	blit
0014.	fe	SPECIAL	yield
0015.	11	PUSHB	[ff]
0017.	e0	USER	get_length
0018.	21	PEEKB	1
0019.	21	PEEKB	1
0020.	80	BINARY	ADD
0021.	11	PUSHB	[ff]
0023.	85	BINARY	AND
0024.	22	PEEKB	2
0025.	22	PEEKB	2
0026.	80	BINARY	ADD
0027.	11	PUSHB	[c0]
0029.	80	BINARY	ADD
0030.	11	PUSHB	[ff]
0032.	85	BINARY	AND
0033.	23	PEEKB	3
0034.	23	PEEKB	3
0035.	80	BINARY	ADD
0036.	11	PUSHB	[b0]
0038.	80	BINARY	ADD
0039.	11	PUSHB	[ff]
0041.	85	BINARY	AND
0042.	23	PEEKB	3
0043.	11	PUSHB	[01]
0045.	81	BINARY	SUB
0046.	23	PEEKB	3
0047.	11	PUSHB	[ff]
0049.	85	BINARY	AND
0050.	23	PEEKB	3
0051.	11	PUSHB	[ff]
0053.	85	BINARY	AND
0054.	74	UNARY	SHL8
0055.	86	BINARY	OR
0056.	22	PEEKB	2
0057.	11	PUSHB	[ff]
0059.	85	BINARY	AND
0060.	74	UNARY	SHL8
0061.	74	UNARY	SHL8
0062.	86	BINARY	OR
0063.	e3	USER	set_pixel
0064.	01	POP	1
0065.	03	POP	3
0066.	71	UNARY	DEC
0067.	60	JNZ	to 18
0070.	01	POP	1
0071.	e4	USER	blit
0072.	fe	SPECIAL	yield
0073.	71	UNARY	DEC
0074.	60	JNZ	to 17
0077.	01	POP	1
0078.	40	JMP	to 15

