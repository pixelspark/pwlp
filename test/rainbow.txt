/* Clear strip */
for(i = get_length) {
	set_pixel(i-1, 0, 0, 0);
};
blit;
yield;

loop {
	for(color = 255) {
		for(i = get_length) {
			r = (color + i) & 0xFF;
			g = (color + i + 0xC0) & 0xFF;
			b = (color + i + 0xB0) & 0xFF;
			set_pixel(i-1, r, g, b);
		};
		blit;
		yield;
	}
}