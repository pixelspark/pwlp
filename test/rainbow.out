000000 000000 000000 000000 000000 000000 000000 000000 000000 000000 
00c0b0 01c1b1 02c2b2 03c3b3 04c4b4 05c5b5 06c6b6 07c7b7 08c8b8 09c9b9 
ffbfaf 00c0b0 01c1b1 02c2b2 03c3b3 04c4b4 05c5b5 06c6b6 07c7b7 08c8b8 
febeae ffbfaf 00c0b0 01c1b1 02c2b2 03c3b3 04c4b4 05c5b5 06c6b6 07c7b7 
fdbdad febeae ffbfaf 00c0b0 01c1b1 02c2b2 03c3b3 04c4b4 05c5b5 06c6b6 
fcbcac fdbdad febeae ffbfaf 00c0b0 01c1b1 02c2b2 03c3b3 04c4b4 05c5b5 
